microservices-connector/internal/controller/gmconnector_controller.go

```diff
@@ -431,6 +458,135 @@ func (r *GMConnectorReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return ctrl.Result{}, nil
 }
 
+// get the svc NamespaceName from the svcCfg and replace the ones in the original config map
+func preprocessUserConfigmap(ns string, hwType string) error {
+	var cfgFile string
+	var embdManifest string
+	var rerankManifest string
+	var tgiManifest string
+	var adjustFile string
+	if hwType == xeon {
+		cfgFile = yaml_dir + "/qna_configmap_xeon.yaml"
+		embdManifest = yaml_dir + tei_embedding_service_yaml
+		rerankManifest = yaml_dir + tei_reranking_service_yaml
+		tgiManifest = yaml_dir + tgi_service_yaml
+		adjustFile = yaml_dir + "/qna_configmap_xeon_adj.yaml"
+	} else if hwType == gaudi {
+		cfgFile = yaml_dir + "/qna_configmap_gaudi.yaml"
+		embdManifest = yaml_dir + tei_embedding_gaudi_service_yaml
+		rerankManifest = yaml_dir + tei_reranking_service_yaml
+		tgiManifest = yaml_dir + tgi_gaudi_service_yaml
+		adjustFile = yaml_dir + "/qna_configmap_gaudi_adj.yaml"
+	} else {
+		return fmt.Errorf("unexpected hardware type %s", hwType)
+	}
+	yamlData, err := os.ReadFile(cfgFile)
+	if err != nil {
+		fmt.Printf("failed to read %s : %v\n", cfgFile, err)
+	} else {
+		fmt.Printf("adjust config: %s\n", cfgFile)
+		// Unmarshal the YAML data into a map
+		var yamlMap map[string]interface{}
+		err = yaml2.Unmarshal(yamlData, &yamlMap)
+		if err != nil {
+			return fmt.Errorf("failed to unmarshal YAML data: %v", err)
+		}
+		cfgmapName := yamlMap["metadata"].(map[interface{}]interface{})["name"].(string) + "-" + hwType
+		yamlMap["metadata"].(map[interface{}]interface{})["name"] = cfgmapName
+		if data, ok := yamlMap["data"].(map[interface{}]interface{}); ok {
+			// Update the value of "TEI_EMBEDDING_ENDPOINT" field
+			if _, ok := data["TEI_EMBEDDING_ENDPOINT"].(string); ok {
+				svcName, port, err := getServiceDetails(embdManifest)
+				if err == nil {
+					data["TEI_EMBEDDING_ENDPOINT"] = fmt.Sprintf("http://%s.%s.svc.cluster.local:%d", svcName, ns, port)
```

Comment:
We may need to extract all these environment variables, then handle them together.